lib/api-helper/project/withProject.ts

```diff
@@ -18,29 +17,99 @@ import { ProjectListenerInvocation } from "./../../api/listener/ProjectListener"
 import {
     CloneOptions,
     GitProject,
+    toStringArray,
 } from "@atomist/automation-client";
+import { SpawnSyncOptions } from "child_process";
 import { ExecuteGoalResult } from "../../api/goal/ExecuteGoalResult";
 import {
     ExecuteGoal,
     GoalInvocation,
 } from "../../api/goal/GoalInvocation";
+import { ProgressLog } from "../../spi/log/ProgressLog";
+import {
+    execPromise,
+    ExecPromiseResult,
+    spawnAndLog,
+    SpawnLogOptions,
+    SpawnLogResult,
+} from "../misc/child_process";
+import { ProjectListenerInvocation } from "./../../api/listener/ProjectListener";
+
+/**
+ * Convenience access to running child processes in the context of a local project
+ */
+export interface ChildProcessOnProject {
+
+    /**
+     * Spawn a child process, by default setting cwd to the directory of the local
+     * project and using the progressLog of GoalInvocation as logger.
+     * See spawnAndLog for more details.
+     * @param {string} cmd
+     * @param {string | string[]} args
+     * @param {SpawnLogOptions} opts
+     * @param {ProgressLog} log
+     * @returns {Promise<SpawnLogResult>}
+     */
+    spawn(cmd: string, args?: string | string[], opts?: SpawnLogOptions, log?: ProgressLog): Promise<SpawnLogResult>;
```

Comment:
It's strange to have a different order of arguments that the functions they are wrapping. That said, the last argument is not necessary as `SpawnLogOptions` already has an optional `log` property inherited from `SpawnPromiseOptions`, although maybe it should really be required and upgraded to a `ProgressLog`, it's currently a `WritableLog`. That would remove the weird argument ordering you commented on a few day ago. We can discuss.

I strongly disagree with making `args` a `string | string[]`. The former is just asking for trouble, and has proven to be extremely problematic on MS Windows.

Reply:
The `progressLog` argument is optional in this context and can go last because we have it from the goal invocation. So the order has to change unfortunately.

Reply:
I recant my objections to the `args` since we are not splitting the string on whitespace to create an array.

I still think we can remove the last argument if we improve the sdm child_process stuff.

Reply:
that would certainly be great!